docs/specs/multi-registry/multi_registry.md

```diff
@@ -0,0 +1,366 @@
+
+> Note: A resolved registry is self-contained and does not include any complex constructs
+> like `imports`, `ref`, `extends`, etc. Their **structure is less subject to change**, making them good
+> candidates for publication and easier to consume.
+
+The following properties are proposed for a resolved registry:
+
+- Resolved registry must be easy to consume and to publish
+  - Accessible via a URL.
+  - Self-contained, i.e. a single file.
+  - No `ref`, no `extends`, no `imports`, no alias, no other complex constructs.
+  - Yaml or JSON format so resolved registries can be easily consumed by any tool.
+- Optional lineage section.
+
+The structure of resolved registries is less subject to change. We can always add advanced support for embedding,
+more complex inheritance mechanisms, etc. The resolved registry format will not change because all these complex
+constructs will be removed during the resolution process. This property makes resolved registries good candidates
+for publication and easier to consume.
+
+The content of a resolved registry depends on the:
+- The semantic convention files composing the registry to resolve.
+- The registries imported.
+- The configuration specified during the resolution process.
+  - Include all the entities of the imported registries
+  - Include only the referenced entities of the imported registries.
+
+More specifically, a resolved registry contains:
+- All the attributes registry specified locally in the registry.
+- All the groups specified locally in the registry.
+- All the attributes and groups imported but not re-exported locally are not included in the resolved registry. A
+  re-exported entity is an entity that is imported and referenced in the local registry with some overriding.
+
+Open Questions:
+
+- Do we keep track of the imported registries in the resolved registry? If yes, how? Lineage?
+- Can we leverage the attribute deduplication mechanism to simplify the merging of imported registries? ToDo -> Explain
+  attribute deduplication mechanism.
+- Can we extend the deduplication mechanism to the signals? 
+- Materialized resolved registry (what see the jq, template and policy engines) vs Published resolved registry.
+  - Materialized Resolved Registry: This is what the jq, template and policy engines see. In this format there are
+  no deduplication of declaration. This format is not meant to be published.
+  - Published Resolved Registry: In this format, the deduplication of declaration is automatically done by Weaver. 
+  This format is meant to be published.
+
+## Priorities
+
+TBD
+
+## General Open Questions
+
+- What about introducing a new type of semconv file that will let end-users define global overrides and global redact
+  directives? For example, the requirement level of attributes such as `client.address` or `server.address` will be
+  better defined by the end-user than by the library author, or a vendor. A similar approach could be used for redact
```

Comment:
future concern: we should be crystal clear that user modifying requirement level in their registry has no effect on instrumentation libraries or produced telemetry.

Reply:
Yes, in a general context. I will add this point to the document. However, I also believe that Weaver can enable the development of a new type of type-safe Client SDK, where a change in the `requirement_level` would have a direct impact on what is reported by the applications using this SDK. Personally, I am a strong believer in this approach.

Reply:
I love the idea, but it needs all instrumentation libraries to change. 

I.e. if my library reports `http.request.header.foo` as opt-in (or doesn't report it at all), it's not in the OTel SDK power to make my instrumentation report it.

Reply:
I don’t think we need to change anything in the current instrumentation libraries. In my view, we can follow a two-step approach. The first step involves proposing a library that offers a type-safe API on top of the existing SDK clients. Those who prefer the current approach can continue using the current SDK client as they always have. Those who want the pure type-safe API approach will only interact with this additional layer. It is even conceivable that a user might want to adopt a mixed approach, using one or the other APIs depending on the need.

In step 2, the idea is to no longer base the type-safe API on the SDK clients, but to provide a fully optimized version by taking advantage of the fact that a type-safe API doesn’t need all the overhead of a generic approach. Hashmaps, abstraction layers, etc., can either be optimized or completely removed.

The advantage of this two-step approach is that we can test the type-safe API approach with the community at a lower cost. I already have a proof of concept for step 1 in Rust.